src/Grapevine/Server/Router.cs

```diff
@@ -531,35 +529,44 @@ public void Route(object state)
             var context = state as IHttpContext;
             if (context == null) return;
```

Comment:
http://stackoverflow.com/questions/323558/generic-threadpool-in-net

These guys had a good idea for getting type safety if you were interested at some point, which basically boils down to a utility method...

static Result QueueItem<T>(Action<T> action, T state)
{
// imagine try catch and returning a result if there is an issue
    ThreadPool.QueueUserWorkItem(delegate { action(state); });
return result; //what ever issues might ensue.
}

and you could do all of the checks that you would like to perform in one spot. Like if it's null you could return a result of if it was a failure or whatever. That way on the off chance someone does call it they can decide if it is a real issue or not, or just ignore the result.